src/backtrace.rs

```diff
@@ -0,0 +1,77 @@
+// SPDX-FileCopyrightText: 2023 Rivos Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+/// Get an array of backtrace addresses.
+///
+/// This needs `force-frame-pointers` enabled.
+use core::arch::asm;
+
+pub const MAX_BACKTRACE_ADDRESSES: usize = 10;
+
+extern "C" {
+    static _stack_start: u8;
+    static _stack_end: u8;
+}
+
+#[derive(Copy, Clone)]
+pub(crate) enum BTReturn {
+    Some(usize),
+    None,
+    BadFp(usize),
+    BadAddress(usize),
```

Comment:
Nit: Since both u64 and usize are are 64-bit, using u64 will avoid the requirement for casting.

Reply:
fixed